pipewire: 0.3.25 -> 0.3.26

balsoft opened the pull request:
###### Motivation for this change

A new update is out for pipewire, fixing many bugs and adding bluetooth Absolute Volume.

Changelog: <https://gitlab.freedesktop.org/pipewire/pipewire/-/releases/0.3.26>

This release should be API and ABI compatible with all other 0.3.x releases.

CC @jtojnar @jansol @gebner @ryantm @nglen

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

balsoft committed:
pipewire: 0.3.25 -> 0.3.26

Changelog:
https://gitlab.freedesktop.org/pipewire/pipewire/-/releases/0.3.26

This release should be API and ABI compatible with all other 0.3.x
releases.

jansol reviewed (changes requested):
Please update the default config json files in `nixos/modules/services/desktops/pipewire` as described in the README.md of that directory. Seems there have been some upstream changes to the templates.

balsoft commented:
Tested on NixOS, with something similar to

```
nix-build -A pipewire
systemctl --user stop pipewire.service pipewire.socket pipewire-media-session.service pipewire-pulse.service pipewire-pulse.socket
./result/bin/pipewire &
./result-mediaSession/bin/pipewire-media-session &
./result-pulse/bin/pipewire-pulse &
```

balsoft commented:
@jansol btw, why don't we just take those files from `cfg.package` there?

jtojnar commented:
Yeah, IFD. Ideally, we would have updating those files part of update scripts and checking consistency in `passthru.tests` `on my to-do list`.

jtojnar commented on pkgs/development/libraries/pipewire/0055-pipewire-media-session-path.patch:
```diff
@@ -18,10 +18,10 @@ index 5c4d1af0..7296220f 100644
--systemd_config.set('PW_MEDIA_SESSION_BINARY', join_paths(pipewire_bindir, 'pipewire-media-session'))
-+systemd_config.set('PW_MEDIA_SESSION_BINARY', join_paths(get_option('media-session-prefix'), 'bin/pipewire-media-session'))
+ systemd_config.set('PW_BINARY', pipewire_bindir / 'pipewire')
+ systemd_config.set('PW_PULSE_BINARY', get_option('pipewire_pulse_prefix') / 'bin' / 'pipewire-pulse')
```
To reduce diff noise, we should probably keep 'bin/pipewire-pulse'. Same elsewhere.

balsoft committed:
fixup! pipewire: 0.3.25 -> 0.3.26

balsoft committed:
pipewire, nixos/pipewire: improve documentation

- README.md in the NixOS module is outdated, since now we use the json
files that are generated with spa-json-dump;
- Mention the fact that the NixOS module has to be updated in the
package comments.

balsoft committed:
fixup! pipewire: 0.3.25 -> 0.3.26

jansol commented on nixos/modules/services/desktops/pipewire/README.md:
```diff
@@ -1,6 +1,6 @@
 # Updating
 
 1. Update the version & hash in pkgs/development/libraries/pipewire/default.nix
-2. run `nix build -f /path/to/nixpkgs/checkout pipewire pipewire.mediaSession`
```
This was probably an accidental deletion? The mediaSession output contains the config templates for the session manager.

balsoft replied:
The `mediaSession` output is built with just `nix build -f /path/to/nixpkgs pipewire`:
```
$ rm result*
$ nix build -f . pipewire
$ echo result*
result result-dev result-doc result-installedTests result-jack result-lib result-mediaSession result-pulse
```

jansol replied:
I only get `result` if I do that. Might be down to some nix option in configuration.nix?

ymatsiuk commented:
Thanks @balsoft
quick note/observation: once applied this PR the hardware volume controls stopped working (WH-1000XM3), otherwise everything looks good so far.

talyz committed:
pipewire: Fix tests

balsoft committed:
fixup! pipewire, nixos/pipewire: improve documentation

SuperSandro2000 commented:
@ofborg eval

hmenke commented:
Could you please incorporate https://github.com/NixOS/nixpkgs/pull/120833/commits/d393845335ed427f38144ce41aea4153120b9de8 from the duplicate #120833?

balsoft commented:
The "duplicate" PR does essentially the same thing but also adds an update script, which is a great improvement. To prevent unnecessary duplication, I'm closing this one.
